recipe-client-addon/lib/SandboxManager.jsm

```diff
@@ -65,6 +65,43 @@ this.SandboxManager = class {
       }
     });
   }
+
+  /**
+   * Wraps a function that returns a Promise from a privileged (e.g. chrome)
```

Comment:
"It's complicated". "system principal" is typically what we understand "privileged" or "chrome privileged" to mean, and there's only 1 of those.

However, the code here would also use for 2 non-privileged scopes (so if you had sandboxed bar.com code that cared about a promise from foo.com). So maybe just 'privilege' isn't necessarily the best term here, although of course really the only consumers here care about privileged code <-> sandboxed code interaction...